subprojects/core/src/main/java/org/gradle/execution/plan/Node.java

```diff
@@ -228,12 +237,32 @@ public boolean allDependenciesComplete() {
     }
 
     public boolean allDependenciesSuccessful() {
-        for (Node dependency : dependencySuccessors) {
-            if (!dependency.isSuccessful()) {
-                return false;
-            }
-        }
-        return true;
+        return dependencySuccessors.stream().allMatch(this::shouldContinueExecution);
+    }
+
+    /**
+     * This {@link Node} may continue execution if the successor Node was successful, or if non-successful when two specific criteria are met:
+     * <ol>
+     *     <li>The successor node failure is a "verification failure"</li>
+     *     <li>The relationship to the successor Node is via task output/task input wiring, not an explicit dependsOn relationship (which are discouraged)</li>
+     * </ol>
+     *
+     * @param dependency a successor node in the execution plan
+     * @return true if the successor task was successful, or failed but a "recoverable" verification failure and this Node may continue execution; false otherwise
+     * @see <a href="https://github.com/gradle/gradle/issues/18912">gradle/gradle#18912</a>
+     */
+    protected boolean shouldContinueExecution(Node dependency) {
+        return dependency.isSuccessful() || dependency.isVerificationFailure() && !isLifecycleDependency(dependency);
```

Comment:
```suggestion
        return dependency.isSuccessful() || (dependency.isVerificationFailure() && !isLifecycleDependency(dependency));
```

Nit: I like the brakets for better readability.

Reply:
No worries - I considered the same.